python-avd/pyavd/_eos_designs/structured_config/flows/__init__.py

```diff
@@ -124,70 +109,96 @@ def _enable_sflow(self) -> bool:
-        global_settings = self.inputs.flow_tracking_settings.hardware if tracker_type == "hardware" else self.inputs.flow_tracking_settings.sampled
-        flow_tracking[tracker_type] = global_settings._as_dict()
-        if tracker_type == "sampled":
-            # asdict does not contain default values so we need to insert the default sample.
-            # TODO: consider if asdict should include defaults.
-            flow_tracking[tracker_type]["sample"] = global_settings.sample
-
-        filtered_trackers = []
-        for tracker_name in natural_sort(configured_trackers):
-            """
-            We allow overriding the default flow tracker name, so if user has configured a tracker
-            with the default tracker name, then we just use that, if not, we create a default config
-            """
-            default_tracker = next(iter(EosDesigns.FlowTrackingSettings().trackers))
-            if tracker_name not in self.inputs.flow_tracking_settings.trackers:
-                if tracker_name == default_tracker.name:
-                    tracker = default_tracker
-                else:
-                    msg = f"{tracker_name} is being used for one of the interfaces, but is not configured in flow_tracking_settings"
-                    raise AristaAvdInvalidInputsError(msg)
-            else:
-                tracker = self.inputs.flow_tracking_settings.trackers[tracker_name]
+    def _set_hardware_flow_tracking(self) -> None:
+        """Set the structured configuration for hardware flow tracking if any interface is configured."""
+        all_interfaces = chain(
+            self.structured_config.ethernet_interfaces, self.structured_config.port_channel_interfaces, self.structured_config.dps_interfaces
+        )
 
-            filtered_trackers.append(self.resolve_flow_tracker_by_type(tracker))
+        trackers: set[str] = {interface.flow_tracker.hardware for interface in all_interfaces if interface.flow_tracker.hardware}
+        if not trackers:
+            return
 
-        flow_tracking[tracker_type]["trackers"] = filtered_trackers
-        flow_tracking[tracker_type]["shutdown"] = False
+        self.structured_config.flow_tracking.hardware.record.format_ipfix_standard_timestamps_counters = (
+            self.inputs.flow_tracking_settings.hardware.record.format_ipfix_standard_timestamps_counters
+        )
+        self.structured_config.flow_tracking.hardware.shutdown = False
+
+        # Validate and configure trackers
+        for tracker_name in natural_sort(trackers):
+            config = self._get_tracker_input_config(tracker_name)
+            self.structured_config.flow_tracking.hardware.trackers.append_new(
+                name=config.name,
+                record_export=config.record_export._cast_as(EosCliConfigGen.FlowTracking.Hardware.TrackersItem.RecordExport),
+                exporters=config.exporters._cast_as(EosCliConfigGen.FlowTracking.Hardware.TrackersItem.Exporters),
+            )
```

Comment:
Please check if we can modify the schemas to be $ref to cli gen so we don't need this

Reply:
same as above..